Classes/Notifications/NotificationsViewController.swift

```diff
@@ -56,6 +56,28 @@ FlatCacheListener {
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        if NotificationClient.readOnOpen(){
+            self.readOnOpen()
+        }
+        
+    }
+    
+   private func readOnOpen(){
+        var count = 0
+        let showAll = self.showAll
+        for id in notificationIDs {
```

Comment:
I would do this loop a bit shorter: 

```swift
    for id in notificationIDs where showAll || !client.notificationOpened(id: id) {
        count += 1
    }

```

What do you think?

Reply:
Or even `client.notificationOpened(id: id) == false`, to increase readability.